OpenRA.Game/Map/MapCache.cs

```diff
@@ -122,6 +122,48 @@ public void LoadMaps()
 			}
 		}
 
+		public IEnumerable<Map> EnumerateMapsWithoutCaching(MapClassification classification = MapClassification.System)
+		{
+			// Utility mod that does not support maps
+			if (!modData.Manifest.Contains<MapGrid>())
+				yield break;
+
+			// Enumerate map directories
+			foreach (var kv in modData.Manifest.MapFolders)
+			{
+				MapClassification packageClassification;
+				if (!Enum.TryParse(kv.Value, out packageClassification))
+					continue;
+
+				if (packageClassification != classification)
+					continue;
+
+				var name = kv.Key;
+				var optional = name.StartsWith("~", StringComparison.Ordinal);
+				if (optional)
+					name = name.Substring(1);
+
+				// Don't try to open the map directory in the support directory if it doesn't exist
+				if (name.StartsWith("^", StringComparison.Ordinal))
+				{
+					// Assume that the path is a directory if there is not an existing file with the same name
+					var resolved = Platform.ResolvePath(name);
+					if (!File.Exists(resolved))
```

Comment:
The support directory is not a file, so this always fails.  This needs to also check `Directory.Exists`.  The comment also needs to be corrected.